src/test/ReactTestUtils.js

```diff
@@ -178,9 +178,6 @@ var ReactTestUtils = {
    * @return {array} an array of all the matches.
    */
   scryRenderedDOMComponentsWithClass: function(root, classNames) {
-    if (!Array.isArray(classNames)) {
```

Comment:
Inside of `scryRenderedDOMComponentsWithClass`, that takes in the `root` (instance of component) as the first argument, it calls `ReactTestUtils.findAllInRenderedTree(root)` which checks if `root` is a valid instance of a react component.

Reference: https://github.com/ipeters90/react/blob/testutils-invalid-args/src/test/ReactTestUtils.js#L168-L171

 So if someone calls `scryRenderedDOMComponentsWithClass` with no arguments or with an invalid `root` argument, it will throw an appropriate error (`Invariant Violation: findAllInRenderedTree(...): instance must be a composite component`). That seems sufficient to me. Do you think there's additional checks needed ? @sarbbottam